js/debug-bar-queries.dev.js

```diff
@@ -0,0 +1,44 @@
+jQuery( document ).ready( function() {
+
+	var elapsedSort = function( a, b ) {
+
+		// Account for elapsed times containing commas.
+		var parsedA = parseFloat( a.dataset.ms_elapsed.replace( /,/g, '' ) );
+		var parsedB = parseFloat( b.dataset.ms_elapsed.replace( /,/g, '' ) );
+
+		if ( parsedA < parsedB ) {
+			return 1;
+		}
+		if ( parsedA > parsedB ) {
+			return -1;
+		}
+		return 0;
+	};
+
+	var ordinalSort = function( a, b ) {
+		var parsedA = parseInt( a.dataset.count.replace( /,/g, '' ), 10 );
+		var parsedB = parseInt( b.dataset.count.replace( /,/g, '' ), 10 );
+
+		if ( parsedA > parsedB ) {
+			return 1;
+		}
+		if ( parsedA < parsedB ) {
+			return -1;
+		}
+		return 0;
```

Comment:
Line 9-15 and 22-28 are duplicate code and could be refactored.
What about adding a helper function like:
```js
function sortQueries( $valueA, $valueB ) {
	if ( parsedA > parsedB ) {
		return 1;
	}
	if ( parsedA < parsedB ) {
		return -1;
	}
	return 0;
}
```

Reply:
Though these two pieces of code are very similar, but they are not actually the same. The comparison and return values are different.

The first instance is sorting by elapsed time so it is doing a sort of largest (slowest) down to smallest (quickest).
The second instance is an ordinal sort, sorting from smallest (first) to largest (last).

Reply:
The logic in the two blocks is actually not duplicative.
The logic in the first block is for elapsed time, sorting largest (slowest) down to smallest (fastest).
The logic in the second block returns the ordinal order, smallest (first) to largest (last).

The operators and return values do not match due to this difference. That is why I did not abstract that out.

Reply:
I completely missed that. Thanks for pointing that out.
I think it could still be refactored in that case, with one closure using return sortQueries(...); and the other return - sortQueries(...);, but I'll leave it up to you.